File: chrestomathy_code/tasks/math/factorial/factorial_iter.py
import sys


def factorial(n):
    if n < 0:
        raise ValueError
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result


n = int(sys.argv[1])
try:
    print(factorial(n))
except ValueError:
    print('_invalid_')

File: chrestomathy_code/tasks/math/factorial/factorial_iter.c
#include <stdio.h>
#include <stdlib.h>

#define INV_CODE -1

unsigned long factorial (long n) {
    if (n < 0) {
        return INV_CODE;
    }
    unsigned long result = 1;
    for (int i = 2; i <= n; ++i) {
        result *= i;
    }
    return result;
}

int main (int argc, char *argv[]) {
    int n = atoi(argv[1]);
    unsigned long res = factorial(n);
    if (res == INV_CODE) {
        printf("_invalid_");
    } else {
        printf("%ld", res);
    }
    return 0;
}
